vnext/Shared/InspectorPackagerConnection.cpp

```diff
@@ -0,0 +1,233 @@
+
+namespace facebook {
+namespace react {
+IDestructible::~IDestructible() {}
+} // namespace react
+} // namespace facebook
+
+namespace Microsoft::ReactNative {
+
+namespace {
+
+struct InspectorProtocol {
+  static constexpr std::string_view Message_PAGEID = "pageId";
+  static constexpr std::string_view Message_PAYLOAD = "payload";
+
+  static constexpr std::string_view Message_eventName_wrappedEvent = "wrappedEvent";
+  static constexpr std::string_view Message_eventName_getPages = "getPages";
+  static constexpr std::string_view Message_eventName_connect = "connect";
+  static constexpr std::string_view Message_eventName_disconnect = "disconnect";
+
+  static constexpr std::string_view Message_EVENT = "event";
+
+  enum class EventType { GetPages, WrappedEvent, Connect, Disconnect };
+
+  static EventType getEventType(const folly::dynamic &messageFromPackager) {
+    std::string event = messageFromPackager.at(InspectorProtocol::Message_EVENT).getString();
+    if (0 == event.compare(Message_eventName_getPages)) {
+      return EventType::GetPages;
+    }
+
+    if (0 == event.compare(Message_eventName_wrappedEvent)) {
+      return EventType::WrappedEvent;
+    }
+
+    if (0 == event.compare(Message_eventName_connect)) {
+      return EventType::Connect;
+    }
+
+    if (0 == event.compare(Message_eventName_disconnect)) {
+      return EventType::Disconnect;
+    }
+
+    assert(false && "Unknown event!");
+    std::abort();
+  }
+
+  static folly::dynamic constructResponseForPackager(EventType eventType, folly::dynamic &&payload) {
+    folly::dynamic response = folly::dynamic::object;
+
+    switch (eventType) {
+      case EventType::GetPages:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_getPages;
+        break;
+      case EventType::WrappedEvent:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_wrappedEvent;
+        break;
+      case EventType::Connect:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_connect;
+        break;
+      case EventType::Disconnect:
+        response[InspectorProtocol::Message_EVENT] = InspectorProtocol::Message_eventName_disconnect;
+        break;
+      default:
+        assert(false && "Unknown event Type.");
+        std::abort();
+    }
+
+    response[InspectorProtocol::Message_PAYLOAD] = std::move(payload);
+    return response;
+  }
+
+  static folly::dynamic constructGetPagesResponsePayloadForPackager(
+      const std::vector<facebook::react::InspectorPage> &pages) {
+    folly::dynamic payload = folly::dynamic::array;
+    for (const facebook::react::InspectorPage &page : pages) {
+      folly::dynamic pageDyn = folly::dynamic::object;
+      pageDyn["id"] = page.id;
+      pageDyn["title"] = page.title;
+      pageDyn["vm"] = page.vm;
+
+      pageDyn["isLastBundleDownloadSuccess"] = true;
+      pageDyn["bundleUpdateTimestamp"] = "0";
```

Comment:
Nit: I assume the timestamp is used for some caching?
If you know perhaps a comment what effect returning"0" timestamp has.

Reply:
Ah.. This is something i deferred.. Thanks for catching it .. I'm working on it.

Reply:
This is fixed now.